src/openms/source/METADATA/SpectrumMetaDataLookup.cpp

```diff
@@ -246,6 +246,40 @@ namespace OpenMS
     return success;
   }
 
+  bool SpectrumMetaDataLookup::addMissingIMToPeptideIDs(vector<PeptideIdentification>& peptides, const String& filename,
+  bool stop_on_error)
+  {
+    PeakMap exp;
+    SpectrumLookup lookup;
+    bool success = true;
+    for (auto& pep : peptides)
+    {
+      if (lookup.empty())
+      {
+        FileHandler fh;
+        auto opts = fh.getOptions();
+        // speed up reading. We do not need the actual peaks in the spectra
+        opts.setFillData(false);
+        opts.setSkipXMLChecks(true);
+        fh.setOptions(opts);
+        fh.loadExperiment(filename, exp, {FileTypes::MZXML, FileTypes::MZML, FileTypes::MZDATA, FileTypes::MGF}, OpenMS::ProgressLogger::NONE, true, true);
+        lookup.readSpectra(exp.getSpectra());
+      }
+      String spectrum_id = pep.getSpectrumReference();
+      try
+      {
+        Size index = lookup.findByNativeID(spectrum_id);
+        pep.setMetaValue(exp[index].getDriftTimeUnitAsString(), exp[index].getDriftTime());
```

Comment:
@cbielow and @jpfeuffer what do you think? is it better to use a more specific meta value here or something more generic (like the constants::IM)?

Reply:
I see several possibilities for annotating IM values in the ID datastructures:
1. use the more general meta value key "IM" for all types
2. use a specific one (as annotated as drift time unit string)

Note that this code currently annotates:

For spectra containing no IM information, it adds:
<UserParam type="float" name="\<NONE>" value="-1.0"/ 

In Constants.h we have a meta value key that could be used instead:
      /** User parameter name for general ion mobility values (e.g., if not further specified)
              String
      */
      inline const std::string IM = "IM";


I think both options come with pro and cons... 

- How about not annotating any value if IM information is missing?
- Can the caller also quickly find out what kind of IM has been annotated? e.g., otherwise the user might need to look into meta value keys and see if one is a valid IM type

Reply:
>How about not annotating any value if IM information is missing?

Sure, probably the best option
>Can the caller also quickly find out what kind of IM has been annotated?

Should that be a UserParam in the protein_ids or sth?

Reply:
> @cbielow and @jpfeuffer what do you think? is it better to use a more specific meta value here or something more generic (like the constants::IM)?

I'd go generic `constants::IM`, since 1) downstream algorithms may not care about the unit of IM  2) you can tell by looking at only one metavalue if IM is present. Otherwise you'd have to check a few. 
If you must, add the unit as another metavalue (ideally once).